src/convertToObject.js

```diff
@@ -7,6 +7,20 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  const x = sourceString
+    .toString()
+    .split(';')
+    .map((item) => item.toString().split(':'));
+
+  x.map((item) => {
+    if (!(item[1] === undefined)) {
+      result[item[0].trim()] = item[1].trim();
+    }
```

Comment:
[CODE KNOWLEDGE] - Using `map` here is not appropriate because you are not transforming the array, you're using it to iterate over elements. Use `forEach` instead, as it is intended for side effects like modifying an external variable.